### PR TITLE
Added part channel functionality

### PR DESCRIPTION
**Pull request by jteague9**
New feature allowing bots to part channels in the same way that they can join them now.

* **What is the current behavior?**
There is no way currently for a bot to leave channels.

* **Does this PR introduce a breaking change?**
Doesn't impact existing code.

### CODE REVIEW

File: twitchio/websocket.py [hunk 1]
```diff
@@ -540,10 +570,13 @@ def _task_callback(self, data, task):
     async def part_action(self, channel: str, author: str, tags):
         log.debug('ACTION:: PART: %s', channel)
 
-        if author == self.nick:
```
Comment (OrangutanGaming) [inline twitchio/websocket.py]: This shouldn't really be part of this PR

File: twitchio/websocket.py [hunk 2]
```diff
@@ -540,10 +570,13 @@ def _task_callback(self, data, task):
...
+        if author.lower() == self.nick.lower():
```
Comment (OrangutanGaming) [inline twitchio/websocket.py]: This shouldn't really be part of this PR

File: twitchio/websocket.py [hunk 3]
```diff
@@ -283,6 +284,35 @@ def is_connected(self) -> bool:
...
+            An argument list of channels to attempt parting.
+        """
+
+        await asyncio.gather(*[self._part_channel(x) for x in channels])
+
+    async def _part_channel(self, entry):
+        channel = re.sub('[#\s]', '', entry).lower()
+
+        if channel not in self._channel_cache:
+            raise ClientError(f'Request to leave the "{channel}" channel failed. You are not in that channel.')
```
Comment (EvieePy) [inline twitchio/websocket.py]: Should be `Request to leave the channel "{channel}" failed.`

Comment (EvieePy) [reply twitchio/websocket.py]: Same for else where.

File: twitchio/websocket.py (hunk 2 above)
Comment (EvieePy) [inline twitchio/websocket.py]: It would make more sense to just use .lower() in the constructor.

Comment (EvieePy) [reply twitchio/websocket.py]: That way self.nick is always in lower case form.

File: twitchio/websocket.py (hunk 3 above)
Comment (jteague9) [reply twitchio/websocket.py]: Changed in 6a1feb1

File: twitchio/websocket.py (hunk 2 above)
Comment (jteague9) [reply twitchio/websocket.py]: Fixed in aeaf2cf

**Comment by EvieePy** [comment]
Thanks for the PR :)
